setup/src/opts.ts

```diff
@@ -93,6 +114,14 @@ export function getOpts(
   const stackSetupGhc = (inputs['stack-setup-ghc'] || '') !== '';
   const stackEnable = (inputs['enable-stack'] || '') !== '';
   const matcherDisable = (inputs['disable-matcher'] || '') !== '';
+  // Andreas, 2023-01-05, issue #29:
+  // 'cabal-update' has a default value, so we should get a proper boolean always.
+  // Andreas, 2023-01-06: This is not true if we use the action as a library.
+  // Thus, need to patch with default value here.
+  const cabalUpdate = parseYAMLBoolean(
```

Comment:
Hi, your solution is more correct than the actual behaviour.
However, i think the codebase should be consistently "incorrect" in this case.  It would be even more confusing to users having two ways to set options and being both opposites (setting "false")

So i think it would be better to keep actual semantics for a non breaking pr (using `skip-cabal-update` or `disable-cabal-update` and actual broken boolean behaviour) and in another breaking pr fix boolean semantics for all options at once. See https://github.com/actions/runner/issues/1483

But i also see the advantages of being partially correct so as you consider.

Reply:
Thanks for the re-review, @jneira !

Indeed, sticking to the current handling of "boolean" flags we would need a negative one like `skip-cabal-update`.  Personally, I prefer positive flags (maybe because as a constructivist I shun double negation).  So if we introduce `skip-cabal-update` now it would stick, because even if we fix the handling of the boolean flags, we would want to do it in a backwards compatible way.  In particular, if workflows only used `true` and `false` for the "boolean" flags, they would work the same after the fix.  Only workflows that e.g. used `disable-matcher: foo` to disable the matcher would be broken by the fix.

I see three ways to go on from here:
1. Merge and release this PR and live with a slightly confusing separation of `"boolean"` and `boolean` flags.  Fix the flag syntax soonish.
2. Release with "boolean" `skip-cabal-update` flag.  No urgency to fix the flag syntax.
3. Fix the flag syntax first and postpone this PR until after that.

Depending on developer availability, 3 and the second part of 1 could take some time.  Alt 2 and the first part of 1 could move quickly since it is little work.

Reply:
good summary of alternatives, I am slightly in favour of the conservative one, and if the pr is meged in its actual state I would add a big warning in the readme and in the code itself noting the opposite semantics of the new option, comparing it with old ones explicitly

Reply:
@newhoggy : Since you originally opened issue #29, do you have a preference when it comes to `cabal-update: false` or `skip-cabal-update: true`?

Reply:
I don't have a preference.  Either is fine.

Reply:
to be clear: i also deeply *hate* negative flags and it is a good argument to do it right in the first place but with big warnings everywhere please